src/Ethereum/ABI/Array.cpp

```diff
@@ -133,4 +133,44 @@ std::string ParamArray::getExtraTypes(std::vector<std::string>& ignoreList) cons
     return (proto != nullptr) ? proto->getExtraTypes(ignoreList) : "";
 }
 
+void ParamArrayFix::encode(Data& data) const {
+    this->_params.encode(data);
+}
+
+bool ParamArrayFix::decode(const Data& encoded, size_t& offset_inout) {
+    std::size_t origOffset = offset_inout;
+    auto res = this->_params.decode(encoded, offset_inout);
+    // padding
+    offset_inout = origOffset + ValueEncoder::paddedTo32(offset_inout - origOffset);
+    return res;
+}
+
+bool ParamArrayFix::setValueJson(const std::string& value) {
+    auto valuesJson = json::parse(value, nullptr, false);
+    if (valuesJson.is_discarded() || !valuesJson.is_array() || _params.getCount() < valuesJson.size()) {
```

Comment:
Size comparison is not strict here, and in fact reversed. If more elements are provided, that is rejected, while if less, that's accepted.
I suggest making the condition strict, change `<` into `!=`.

Reply:
Thanks, I had a doubt here if we should accept partial overriding!